To: vim_dev@googlegroups.com
Subject: Patch 7.3.309
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.309 (after 7.3.307)
Problem:    Warnings for pointer types.
Solution:   Change PySliceObject to PyObject.
Files:	    src/if_python3.c


*** ../vim-7.3.308/src/if_python3.c	2011-09-14 15:01:54.000000000 +0200
--- src/if_python3.c	2011-09-14 15:30:45.000000000 +0200
***************
*** 1030,1036 ****
      } else if (PySlice_Check(idx)) {
  	Py_ssize_t start, stop, step, slicelen;
  
! 	if (PySlice_GetIndicesEx((PySliceObject *)idx,
  	      (Py_ssize_t)((BufferObject *)(self))->buf->b_ml.ml_line_count+1,
  	      &start, &stop,
  	      &step, &slicelen) < 0) {
--- 1030,1036 ----
      } else if (PySlice_Check(idx)) {
  	Py_ssize_t start, stop, step, slicelen;
  
! 	if (PySlice_GetIndicesEx((PyObject *)idx,
  	      (Py_ssize_t)((BufferObject *)(self))->buf->b_ml.ml_line_count+1,
  	      &start, &stop,
  	      &step, &slicelen) < 0) {
***************
*** 1054,1060 ****
      } else if (PySlice_Check(idx)) {
  	Py_ssize_t start, stop, step, slicelen;
  
! 	if (PySlice_GetIndicesEx((PySliceObject *)idx,
  	      (Py_ssize_t)((BufferObject *)(self))->buf->b_ml.ml_line_count+1,
  	      &start, &stop,
  	      &step, &slicelen) < 0) {
--- 1054,1060 ----
      } else if (PySlice_Check(idx)) {
  	Py_ssize_t start, stop, step, slicelen;
  
! 	if (PySlice_GetIndicesEx((PyObject *)idx,
  	      (Py_ssize_t)((BufferObject *)(self))->buf->b_ml.ml_line_count+1,
  	      &start, &stop,
  	      &step, &slicelen) < 0) {
***************
*** 1142,1148 ****
      } else if (PySlice_Check(idx)) {
  	Py_ssize_t start, stop, step, slicelen;
  
! 	if (PySlice_GetIndicesEx((PySliceObject *)idx,
  		((RangeObject *)(self))->end-((RangeObject *)(self))->start+1,
  		&start, &stop,
  		&step, &slicelen) < 0) {
--- 1142,1148 ----
      } else if (PySlice_Check(idx)) {
  	Py_ssize_t start, stop, step, slicelen;
  
! 	if (PySlice_GetIndicesEx((PyObject *)idx,
  		((RangeObject *)(self))->end-((RangeObject *)(self))->start+1,
  		&start, &stop,
  		&step, &slicelen) < 0) {
***************
*** 1164,1170 ****
      } else if (PySlice_Check(idx)) {
  	Py_ssize_t start, stop, step, slicelen;
  
! 	if (PySlice_GetIndicesEx((PySliceObject *)idx,
  		((RangeObject *)(self))->end-((RangeObject *)(self))->start+1,
  		&start, &stop,
  		&step, &slicelen) < 0) {
--- 1164,1170 ----
      } else if (PySlice_Check(idx)) {
  	Py_ssize_t start, stop, step, slicelen;
  
! 	if (PySlice_GetIndicesEx((PyObject *)idx,
  		((RangeObject *)(self))->end-((RangeObject *)(self))->start+1,
  		&start, &stop,
  		&step, &slicelen) < 0) {
*** ../vim-7.3.308/src/version.c	2011-09-14 15:39:26.000000000 +0200
--- src/version.c	2011-09-14 15:40:17.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     309,
  /**/

-- 
Never enter the boss's office unless it's absolutely necessary.  Every boss
saves one corner of the desk for useless assignments that are doled out like
Halloween candy to each visitor.
				(Scott Adams - The Dilbert principle)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
